cirq-core/cirq/ops/moment_test.py

```diff
@@ -617,3 +617,41 @@ def test_transform_qubits():
     assert original.transform_qubits(lambda q: cirq.GridQubit(10 + q.x, 20)) == modified
     with pytest.raises(TypeError, match='must be a function or dict'):
         _ = original.transform_qubits('bad arg')
+
+
+def test_single_gate_no_gates():
+    assert cirq.Moment().get_single_gate_from_moment() is None
+
+
+def test_single_gate_same_gates():
+    a = cirq.H(cirq.NamedQubit('a'))
+    b = cirq.H(cirq.NamedQubit('b'))
+    c = cirq.H(cirq.NamedQubit('c'))
+    assert cirq.Moment([a, b, c]).get_single_gate_from_moment() == cirq.H
+
+
+def test_single_gate_different_gates():
+    a = cirq.H(cirq.NamedQubit('a'))
+    b = cirq.X(cirq.NamedQubit('b'))
+    assert cirq.Moment([a, b]).get_single_gate_from_moment() is None
```

Comment:
Maybe add this test to see the problem clearly:

```
    a = cirq.XPowGate(exponent=0.25)(cirq.NamedQubit('a'))
    b =cirq.XPowGate(exponent=0.5)(cirq.NamedQubit('b'))
    assert cirq.Moment([a, b]).get_single_gate_from_moment() is None
```

Reply:
Done.